enhancements/network/on-prem-bgp-networking.md

```diff
@@ -0,0 +1,545 @@
+
+Here is a list of requirements that the customer needs to have in place:
+
+- A BGP router (or multiple) that can peer with the BGP peers on the control plane nodes.
+- ICMP redirects are disabled on the router(s) that will peer with the control plane nodes. This is needed to ensure that the router(s) does not send an ICMP redirect message when traffic destined to the VIPs is received on a node where the VIP is not configured. This is needed to ensure that the traffic is not black-holed in the network.
+
+We will need to document the requirements and the steps to configure the BGP infrastructure, as well as providing as much as possible some validation tools to help the customer to configure the BGP infrastructure correctly.
+
+#### Networking infrastructure maintenance
+
+Router(s) that will peer with the control plane nodes need to be maintained by the customer, but in case of a failure, it's possible that the BGP configuration has to change.
+Therefore, we need to provide a way for the customer to update the BGP configuration of the control plane nodes.
+The customer would update the BGP Configuration via the OpenShift API, and baremetal-runtimecfg which would monitor the Infrastructure object would update the configuration of the control plane nodes and reload FRR.
+
+### Risks and Mitigations
+
+- This design heavily relies on the infrastructure network that is not controlled by OpenShift. Documentation and validations will be provided
+  to ensure that prerequisites are met before a deployment can be done using BGP
+- Performance can be degraded if infrastructure network has routing problems, but routing issues would also involve a general outage, not just control plane VIPs
+- The opinionated configuration for FRR might be too simple for now, but we'll adjust it as we get feedback from customers and field deployments
+
+### Drawbacks
+
+The proposed solution heavily relies on the network infrastructure at the data center level and if something goes wrong
+with BGP, the OpenShift cluster can become very unstable.
+Also, the solution remains complex to implement and corner cases will have to be documented and covered by testing when possible (e.g. day 2 operations).
+
+## Design Details
+
+### Test Plan
+
+Testing will be implemented for OpenStack at first and re-usable by other platforms.
+We'll provide a set of scripts that can install ToR routers and this will be used to replicate
+the network infrastructure that customers could run.
+The jobs will do the following:
+- Create a proxy node that runs FRR and will be the upstream router for the OpenShift cluster
+- Deploy the OpenShift cluster and provides an `install-config.yaml` that has the needed options for BGP
+- Run the desired e2e test suite
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+These following requirements will apply first for OpenStack, and once this has proven to be stable, we'll turn this on for other platforms:
```

Comment:
is there any requirement around OpenStack versions to use?

Reply:
no, I'll add a note.